test-case/check-signal-stop-start.sh

```diff
@@ -61,40 +64,39 @@ setup_kernel_check_point
 func_stop_start_pipeline()
 {
     local i=1
-    while ( [ $i -le $count ] && [ "$(ps -p $pid --no-header)" ] )
+    while [ $i -le $count ]
     do
         # check aplay/arecord process state
-        sof-process-state.sh $cmd >/dev/null
-        if [ $? -ne 0 ]; then
-            "$cmd process is in an abnormal status"
-            kill -9 $pid && wait $pid 2>/dev/null
+        sof-process-state.sh "$pid" >/dev/null || {
+            dloge "$cmd($pid) process is in an abnormal status"
+            kill -9 "$pid"
             exit 1
-        fi
+        }
         dlogi "Stop/start count: $i"
         # stop the pipeline
-        kill -SIGSTOP $pid
+        kill -SIGSTOP "$pid"
         sleep $interval
         # start the pipeline
-        kill -SIGCONT $pid
+        kill -SIGCONT "$pid"
         sleep $interval
-        let i++
+        (( i++ ))
     done
 }
 
 func_pipeline_export "$tplg" "type:$test_mode"
-for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+for idx in $(seq 0 $(( "$PIPELINE_COUNT" - 1 )))
 do
-    channel=$(func_pipeline_parse_value $idx channel)
-    rate=$(func_pipeline_parse_value $idx rate)
-    fmt=$(func_pipeline_parse_value $idx fmt)
-    dev=$(func_pipeline_parse_value $idx dev)
-    pcm=$(func_pipeline_parse_value $idx pcm)
-    pipeline_type=$(func_pipeline_parse_value $idx "type")
-    snd=$(func_pipeline_parse_value $idx snd)
+    channel=$(func_pipeline_parse_value "$idx" channel)
+    rate=$(func_pipeline_parse_value "$idx" rate)
+    fmt=$(func_pipeline_parse_value "$idx" fmt)
+    dev=$(func_pipeline_parse_value "$idx" dev)
+    pcm=$(func_pipeline_parse_value "$idx" pcm)
+    pipeline_type=$(func_pipeline_parse_value "$idx" "type")
+    snd=$(func_pipeline_parse_value "$idx" snd)
 
     dlogi "Testing: run stop/start test on PCM:$pcm,$pipeline_type. Interval time: $interval"
     dlogc "$cmd -D$dev -r $rate -c $channel -f $fmt $dummy_file -q &"
-    $cmd -D$dev -r $rate -c $channel -f $fmt $dummy_file -q &
+    $cmd -D"$dev" -r "$rate" -c "$channel" -f "$fmt" $dummy_file -q &
```

Comment:
Shellcheck didn't care about $dummy_file?

Reply:
shellcheck seems smart enough to find "local" variables and make sure they don't include any problematic character.